### PR TITLE
[DERCBOT-1314] Add documentsRequired Setting to BotRAGConfiguration

### PR DESCRIPTION
**Pull request by scezen**
### Related Tickets
JIRA: **[DERCBOT-1314]**

### Description

This PR introduces a new boolean attribute `documentsRequired` to the `BotRAGConfiguration` and integrates it across the system. The `documentsRequired` attribute determines whether documents are mandatory for processing the RAG chain.

### Key Changes:
1. **Model Updates**:
    - Added `documentsRequired` to [`BotRAGConfiguration`](bot/engine/src/main/kotlin/admin/bot/rag/BotRAGConfiguration.kt) (default: `true`) and [`BotRAGConfigurationDTO`](bot/admin/server/src/main/kotlin/model/BotRAGConfigurationDTO.kt)
    - Ensured backward compatibility by handling cases where `documentsRequired` is absent in the database.

2. **Orchestrator Enhancements**:
    - Updated [`RAGQuery`](gen-ai/orchestrator-client/src/main/kotlin/ai/tock/genai/orchestratorclient/requests/RAGQuery.kt) to include `documentsRequired` as an optional parameter.
    - Modified `__rag_guard` in [`rag_chain.py`](gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/rag_chain.py) to respect the `documentsRequired` setting.

3. **Bot API Modifications**:
    - Updated [`RAGAnswerHandler`](bot/engine/src/main/kotlin/engine/config/RAGAnswerHandler.kt) to fetch the `documentsRequired` setting from the bot configuration and pass it to the RAG query.

### Testing :
- We can block the retrieval of documents by adding a filter to [`pgvectors_params.py`](gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/models/vector_stores/pgvector/pgvector_params.py) as so: 

```
    def to_dict(self):
        return {
            'k': self.k,
            'filter': {'tag1': 'vector stores'},
        }
```
- Then we are free to test if an answer is still provided while no retrieval of document is made.

### CODE REVIEW

**Review by assouktim — CHANGES_REQUESTED** [review]
LGTM, just update the unit tests (python) and everything should be ready.

**Comment by scezen** [comment]
Tests added @assouktim

**Review by assouktim — APPROVED** [review]
Thanks a lot.

**Comment by scezen** [comment]
Fixed the edge case, now handles missing documents without 'no_answer' configuration @Benvii

File: gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/rag_chain.py [hunk 1]
```diff
@@ -238,35 +238,47 @@ def __find_input_variables(template):
...
+        if chain_can_give_no_answer_reply and chain_reply_no_answer:
+            # Expected behavior: chain used 'no_answer' correctly
+            return
+        # Unexpected behavior: chain provided an answer without documents when required
+        message = 'The RAG provides an answer when no document has been found!'
+        __rag_log(level=ERROR, message=message, inputs=inputs, response=response)
+        raise GenAIGuardCheckException(ErrorInfo(cause=message))
+
+    if chain_reply_no_answer and not no_docs_retrieved:
+        # Unexpected behavior: chain replied with 'no_answer' but documents are present
```
Comment (Benvii) [inline gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/rag_chain.py:278]: It's not unexpected sometimes documents found by the retriever doesn't contain the answer to the question. It can happen especially when we don't have any compressor as it will alsways find documents.

**Review by Benvii — CHANGES_REQUESTED** [review]
- [ ] Remove the last condition see my comment.
- [ ] Add a test without the 'no_answer' input to check that it works

**Review by Benvii — APPROVED** [review]
Thanks for the last updates ok for me.
